rnd/autogpt_builder/src/lib/autogpt-server-api/baseClient.ts

```diff
@@ -0,0 +1,325 @@
+import { SupabaseClient } from "@supabase/supabase-js";
+import {
+  Block,
+  Graph,
+  GraphCreatable,
+  GraphUpdateable,
+  GraphMeta,
+  GraphExecuteResponse,
+  NodeExecutionResult,
+  User,
+  UserData,
+  TutorialStepData,
+  PageViewData,
+} from "./types";
+
+export default class BaseAutoGPTServerAPI {
+  private baseUrl: string;
+  private wsUrl: string;
+  private webSocket: WebSocket | null = null;
+  private wsConnecting: Promise<void> | null = null;
+  private wsMessageHandlers: Record<string, Set<(data: any) => void>> = {};
+  private supabaseClient: SupabaseClient | null = null;
+
+  constructor(
+    baseUrl: string = process.env.NEXT_PUBLIC_AGPT_SERVER_URL ||
+      "http://localhost:8000/api",
+    wsUrl: string = process.env.NEXT_PUBLIC_AGPT_WS_SERVER_URL ||
+      "ws://localhost:8001/ws",
+    supabaseClient: SupabaseClient | null = null,
+  ) {
+    this.baseUrl = baseUrl;
+    this.wsUrl = wsUrl;
+    this.supabaseClient = supabaseClient;
+  }
+
+  async createUser(): Promise<User> {
+    return this._request("POST", "/auth/user", {});
+  }
+
+  // Analytics
+  async logCreateUser(userData: UserData): Promise<string> {
+    return this._request("POST", "/analytics/log_new_user", userData);
+  }
+
+  async logTutorialStep(data: TutorialStepData): Promise<string> {
+    return this._request("POST", "/analytics/log_tutorial_step", data);
+  }
+
+  async logPageView(pageViewData: PageViewData): Promise<string> {
+    return this._request("POST", "/analytics/log_page_view", {
+      page_view_data: pageViewData,
+    });
+  }
```

Comment:
Is there a challenge to simplify this into 1 analytics API instead? Would be tricky to keep adding a new API on new metrics.

Reply:
just like the db table.

Reply:
The raw endpoint is that. This just allows some more fun typing things